### ISSUE TITLE
R: Don't install 'recommended' packages

### ISSUE THREAD

**Issue by HenrikBengtsson** [HenrikBengtsson/CBI-software#19]
# Issue

Since we install 'recommended' packages to the system library, users get warnings each time they update their packages;

```r
> update.packages()
Warning: package 'boot' in library '/usr/lib/R/library' will not be updated
Warning: package 'class' in library '/usr/lib/R/library' will not be updated
Warning: package 'KernSmooth' in library '/usr/lib/R/library' will not be updated
Warning: package 'lattice' in library '/usr/lib/R/library' will not be updated
Warning: package 'MASS' in library '/usr/lib/R/library' will not be updated
Warning: package 'Matrix' in library '/usr/lib/R/library' will not be updated
Warning: package 'mgcv' in library '/usr/lib/R/library' will not be updated
Warning: package 'nlme' in library '/usr/lib/R/library' will not be updated
Warning: package 'nnet' in library '/usr/lib/R/library' will not be updated
Warning: package 'survival' in library '/usr/lib/R/library' will not be updated
...
```

This adds quite a bit of noise but it's also confusing and drives users to ignore warnings.

# Decision

Don't install above 'recommended' packages.  Instead, let users install them if they need them.  Also, if they need them because they're needed by another package the user installs, then they'll be automatically installed.

To _not_ install above packages when building R, use:

```
./configure --without-recommended-packages ...
```

**Comment by HenrikBengtsson** [comment]
In order to test our R build, we need to patch `make check`. R Core acknowledges the problem and is open to patches

**Comment by HenrikBengtsson** [comment]
This has been fixed for R 4.2.0. 

For R 4.1.0, we need to install the on order to run the checks, but i guess we could uninstall the before deploying. Maybe it's sufficient to install just some of them, e.g. MASS for sure.

**Comment by HenrikBengtsson** [comment]
Done already for R 4.1.1
